platform/mds/mds/src/main/java/org/motechproject/mds/repository/ComboboxValueRepository.java

```diff
@@ -0,0 +1,50 @@
+package org.motechproject.mds.repository;
+
+import org.motechproject.mds.domain.Entity;
+import org.motechproject.mds.domain.Field;
+import org.motechproject.mds.util.Constants;
+import org.springframework.stereotype.Repository;
+
+import javax.jdo.PersistenceManager;
+import javax.jdo.Query;
+import java.util.List;
+
+/**
+ * Responsible for fetching possible values for a combobox from the database.
+ */
+@Repository
+public class ComboboxValueRepository extends AbstractRepository {
+
+    /**
+     * Retrieves all values for a multi-select combobox given its table name in the database.
+     * Multi-select comboboxes have values stored in a separate table, hence we need the table name.
+     * @param cbTableName the name of the combobox tables
+     * @return all values for the combobox currently in the database
+     */
+    public List<String> getComboboxValuesForCollection(String cbTableName) {
+        PersistenceManager pm = getPersistenceManager();
+
+        // we need to execute sql, since jdoql won't allow multi-value fields in the result
+        // tableName is safe since it comes from the metadata, no need for params
```

Comment:
Not that safe for PostgreSQL, that decides to make all its table names upper case.

```java
org.postgresql.util.PSQLException: BúąD: relacja "mds_testing_bbb" nie istnieje
  Pozycja: 30
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryEx
ecutorImpl.java:2102)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutor
Impl.java:1835)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.ja
va:257)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Stat
ement.java:500)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(Abstract
Jdbc2Statement.java:388)
        at org.postgresql.jdbc2.AbstractJdbc2Statement.executeQuery(AbstractJdbc
2Statement.java:273)
        at org.datanucleus.store.rdbms.datasource.dbcp.DelegatingPreparedStateme
nt.executeQuery(DelegatingPreparedStatement.java:94)
        at org.datanucleus.store.rdbms.datasource.dbcp.DelegatingPreparedStateme
nt.executeQuery(DelegatingPreparedStatement.java:94)
        at org.datanucleus.store.rdbms.ParamLoggingPreparedStatement.executeQuer
y(ParamLoggingPreparedStatement.java:375)
        at org.datanucleus.store.rdbms.SQLController.executeStatementQuery(SQLCo
ntroller.java:552)
        at org.datanucleus.store.rdbms.query.SQLQuery.performExecute(SQLQuery.ja
va:276)
        at org.datanucleus.store.query.Query.executeQuery(Query.java:1833)
        at org.datanucleus.store.query.AbstractSQLQuery.executeWithArray(Abstrac
tSQLQuery.java:342)
        at org.datanucleus.store.query.Query.execute(Query.java:1704)
        at org.datanucleus.api.jdo.JDOQuery.execute(JDOQuery.java:223)
        at org.motechproject.mds.repository.ComboboxValueRepository.getComboboxV
aluesForCollection(ComboboxValueRepository.java:32)
        at org.motechproject.mds.service.impl.ComboboxValueHelper.getAllValuesFo
rCombobox(ComboboxValueHelper.java:65)
        at org.motechproject.mds.service.impl.EntityServiceImpl.toFieldDtos(Enti
tyServiceImpl.java:1386)
        at org.motechproject.mds.service.impl.EntityServiceImpl.getFields(Entity
ServiceImpl.java:929)
        at org.motechproject.mds.service.impl.EntityServiceImpl.getEntityFieldsF
orUI(EntityServiceImpl.java:890)
        at sun.reflect.GeneratedMethodAccessor334.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAcces
sorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflecti
on(AopUtils.java:318)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJo
inpoint(ReflectiveMethodInvocation.java:183)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:150)
        at org.springframework.transaction.interceptor.TransactionInterceptor.in
voke(TransactionInterceptor.java:110)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynami
cAopProxy.java:202)
        at com.sun.proxy.$Proxy63.getEntityFieldsForUI(Unknown Source)
        at sun.reflect.GeneratedMethodAccessor334.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAcces
sorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflecti
on(AopUtils.java:318)
        at org.eclipse.gemini.blueprint.service.importer.support.internal.aop.Se
rviceInvoker.doInvoke(ServiceInvoker.java:56)
        at org.eclipse.gemini.blueprint.service.importer.support.internal.aop.Se
rviceInvoker.invoke(ServiceInvoker.java:60)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.support.DelegatingIntroductionInterceptor.doP
roceed(DelegatingIntroductionInterceptor.java:131)
        at org.springframework.aop.support.DelegatingIntroductionInterceptor.inv
oke(DelegatingIntroductionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.eclipse.gemini.blueprint.service.util.internal.aop.ServiceTCCLInt
erceptor.invokeUnprivileged(ServiceTCCLInterceptor.java:70)
        at org.eclipse.gemini.blueprint.service.util.internal.aop.ServiceTCCLInt
erceptor.invoke(ServiceTCCLInterceptor.java:53)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.eclipse.gemini.blueprint.service.importer.support.LocalBundleCont
extAdvice.invoke(LocalBundleContextAdvice.java:57)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.support.DelegatingIntroductionInterceptor.doP
roceed(DelegatingIntroductionInterceptor.java:131)
        at org.springframework.aop.support.DelegatingIntroductionInterceptor.inv
oke(DelegatingIntroductionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(
ReflectiveMethodInvocation.java:172)
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynami
cAopProxy.java:202)
        at com.sun.proxy.$Proxy176.getEntityFieldsForUI(Unknown Source)```